cgi-bin/CATS/UI/RunDetails.pm

```diff
@@ -418,7 +418,9 @@ sub get_last_verdicts_api {
             $rid && url_f('run_details', rid => $rid),
             ($is_jury_in_contest ? url_function('problem_details',
                 cid => $contest_id, pid => $problem_id, sid => $sid) : ''),
-        ]
+        ];
+        $result->{allowed_problems}->{$_} = $is_jury ||
+            $problem_status != $cats::problem_st_disabled && $state != $cats::st_banned;
```

Comment:
Put this inside result property array above, it is already per-problem.
It is probably time to convert that array into hash, you csan do it in separate commit or separate PR